### ISSUE TITLE
PHP Notice: Undefined variable / PHP Warning: array_key_exists() expects parameter

### ISSUE THREAD

**Issue by midide** [moodleuulm/moodle-block_course_overview_campus#30]
A lot of entries in my php error log

[25-Sep-2018 05:07:28 Europe/Berlin] PHP Notice:  Undefined variable: courseteachers in /var/www/moodle/blocks/course_overview_campus/block_course_overview_campus.php on line 644
[25-Sep-2018 05:07:28 Europe/Berlin] PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /var/www/moodle/blocks/course_overview_campus/block_course_overview_campus.php on line 644

Moodle 3.5.2 (Build: 20180910)
PHP Version 7.0.30-0+deb9u1
Course overview on campus block_course_overview_campus v3.5-r1 2018070200

Great plugin by the way!

**Comment by AnsonArgyris** [comment]
Hello!

I can confirm the error above.

And it is a cool plugin, thank you for the great work!

As debug messages i also get the following:
```
The function assign_print_overview() is now deprecated.

The function chat_print_overview() is now deprecated.

The function choice_print_overview() is now deprecated.

The function forum_print_overview() is now deprecated.

The function forum_filter_user_groups_discussions() is now deprecated.

The function lesson_print_overview() is now deprecated.

The function quiz_print_overview() is now deprecated.

The function scorm_print_overview() is now deprecated.
```
Moodle 3.5.1 (Build: 20180709)
PHP Version 7.0.30-0+deb9u1

**Comment by abias** [comment]
Hi @midide, hi @AnsonArgyris,

thank you for this report. The plugin's main code has gotten quite complicated over the last year and there might be edge case combinations with the plugin's settings which might not be covered. 

I have fixed this debug message within 8d765ce9998e69adb3b6111d95527586cc819986 and will publish a new version in the plugin repo soon.

@AnsonArgyris , regarding the additional debug messages, please have a look at https://github.com/moodleuulm/moodle-block_course_overview_campus/blob/master/README.md#5-course-news where this has been already covered.

Cheers,
Alex

**Comment by AnsonArgyris** [comment]
@abias Thank you for this very quick fix!
